SwedbankPaySDK/Classes/Merchant Backend/Problems.swift

```diff
@@ -0,0 +1,185 @@
+//
+// Copyright 2020 Swedbank AB
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import Alamofire
+
+enum Problems {
+    static func parseProblem(
+        response: AFDataResponse<Data>
+    ) -> SwedbankPaySDK.Problem {
+        do {
+            return try parseProblemOrFail(response: response)
+        } catch {
+            return makeUnexpectedContentProblem(response: response)
+        }
+    }
+    
+    static func makeUnexpectedContentProblem(
+        response: AFDataResponse<Data>
+    ) -> SwedbankPaySDK.Problem {
+        // we should always have response here, but we need some value
+        let status = response.response?.statusCode ?? -1
+        let isClient = (400...499).contains(status)
+        if isClient {
+            return .client(.unexpectedContent(
+                status: status,
+                contentType: response.response?.mimeType,
+                body: response.data
+            ))
+        } else {
+            return .server(.unexpectedContent(
+                status: status,
+                contentType: response.response?.mimeType,
+                body: response.data
+            ))
+        }
+    }
+}
+
+private typealias JsonObject = [String: Any]
+
+private extension JsonObject {
+    var type: String { self["type"] as? String ?? "about:blank" }
+    var title: String? { self["title"] as? String }
+    var status: Int? { self["status"] as? Int }
+    var detail: String? { self["detail"] as? String }
+    var instance: String? { self["instance"] as? String }
+}
+
+
```

Comment:
Codacy found an issue: [Limit vertical whitespace to a single empty line. Currently 2.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-ios/pullRequest?prid=6260585)